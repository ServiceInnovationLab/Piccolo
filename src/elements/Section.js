import styled from 'styled-components';

function theme(props) {
  if(props.dark) {
    return '#2f3045';
  } else if(props.light) {
    return '#d9f8fa';
  } else if(props.grey) {
    return '#eeeeee';
  } else {
    return 'transparent';
  }
}


export const Section = styled.section`
  background: ${props => theme(props)};
  font-family: sans-serif;
  padding: 55px 0;
  text-align: center;

  h2 {
    color: ${props => props.dark ? '#42e2ee' : '#2f3045'};
    margin-bottom: 0;
  }

  h2 ~ p {
    margin-top: 10px;
    color: ${props => props.dark ? '#fff' : '#000'};
    padding-bottom: 35px;
  }

  p ~ hr {
    color: ${props => props.dark ? '#ccc' : '#000'};
    height: 1px;
    border: 0;
    border-top: 1px solid ${props => props.dark ? '#ccc' : '#000'};
  }

  h3 {
    color: ${props => props.dark ? '#fff' : '#2f3045'};
    font-weight: 400;
  }

  h2 ~ h3 {
    margin-top: 0;
  }
`;


