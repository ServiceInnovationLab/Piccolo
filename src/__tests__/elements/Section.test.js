import React from 'react';
import renderer from 'react-test-renderer';
import Section from '../../elements/Section';
import 'jest-styled-components';

const section_dark = renderer.create(<Section dark />).toJSON();
const section_light = renderer.create(<Section light />).toJSON();
const section_grey = renderer.create(<Section grey />).toJSON();

test('Section default styles applied', () => {
  const tree = renderer.create(<Section />).toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule('padding', '55px 0');
  expect(tree).toHaveStyleRule('text-align', 'left', {
    media: '(min-width: 1024px)'
  });
});

test('Dark Section default styles applied', () => {
  expect(section_dark).toMatchSnapshot();
  background(section_dark, '#2f3045');
});

test('Light Section default styles applied', () => {
  expect(section_light).toMatchSnapshot();
  background(section_light, '#d9f8fa');
});

test('Grey Section default styles applied', () => {
  expect(section_grey).toMatchSnapshot();
  background(section_grey, '#ccc');
});

function background(section, color) {
  expect(section).toHaveStyleRule('background', color);
}
