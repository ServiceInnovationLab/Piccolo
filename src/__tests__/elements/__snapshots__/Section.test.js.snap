// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dark Section default styles applied 1`] = `
.c0 {
  font-family: sans-serif;
  padding: 55px 0;
  background: #2f3045;
}

.c0 p:first-child {
  margin-top: 0;
}

.c0 h2 {
  color: #2f3045;
  margin-bottom: 0;
  color: #42e2ee;
}

.c0 h2 ~ p {
  margin-top: 10px;
  color: #000;
  padding-bottom: 35px;
  color: #fff;
}

.c0 p ~ hr {
  color: #000;
  height: 1px;
  border: 0;
  border-top: 1px solid #000;
  color: #ccc;
  border-top: 1px solid #ccc;
}

.c0 h3 {
  color: #2f3045;
  font-weight: 400;
}

.c0 h2 ~ h3 {
  margin-top: 0;
}

@media (min-width:1024px) {
  .c0 {
    text-align: left;
  }
}

<section
  className="c0"
/>
`;

exports[`Grey Section default styles applied 1`] = `
.c0 {
  font-family: sans-serif;
  padding: 55px 0;
  background: #ccc;
}

.c0 p:first-child {
  margin-top: 0;
}

.c0 h2 {
  color: #2f3045;
  margin-bottom: 0;
}

.c0 h2 ~ p {
  margin-top: 10px;
  color: #000;
  padding-bottom: 35px;
}

.c0 p ~ hr {
  color: #000;
  height: 1px;
  border: 0;
  border-top: 1px solid #000;
}

.c0 h3 {
  color: #2f3045;
  font-weight: 400;
}

.c0 h2 ~ h3 {
  margin-top: 0;
}

@media (min-width:1024px) {
  .c0 {
    text-align: left;
  }
}

<section
  className="c0"
/>
`;

exports[`Light Section default styles applied 1`] = `
.c0 {
  font-family: sans-serif;
  padding: 55px 0;
  background: #d9f8fa;
}

.c0 p:first-child {
  margin-top: 0;
}

.c0 h2 {
  color: #2f3045;
  margin-bottom: 0;
}

.c0 h2 ~ p {
  margin-top: 10px;
  color: #000;
  padding-bottom: 35px;
}

.c0 p ~ hr {
  color: #000;
  height: 1px;
  border: 0;
  border-top: 1px solid #000;
}

.c0 h3 {
  color: #2f3045;
  font-weight: 400;
}

.c0 h2 ~ h3 {
  margin-top: 0;
}

@media (min-width:1024px) {
  .c0 {
    text-align: left;
  }
}

<section
  className="c0"
/>
`;

exports[`Section default styles applied 1`] = `
.c0 {
  font-family: sans-serif;
  padding: 55px 0;
}

.c0 p:first-child {
  margin-top: 0;
}

.c0 h2 {
  color: #2f3045;
  margin-bottom: 0;
}

.c0 h2 ~ p {
  margin-top: 10px;
  color: #000;
  padding-bottom: 35px;
}

.c0 p ~ hr {
  color: #000;
  height: 1px;
  border: 0;
  border-top: 1px solid #000;
}

.c0 h3 {
  color: #2f3045;
  font-weight: 400;
}

.c0 h2 ~ h3 {
  margin-top: 0;
}

@media (min-width:1024px) {
  .c0 {
    text-align: left;
  }
}

<section
  className="c0"
/>
`;
